ui/src/app/views/account-creation-modals/component.tsx

```diff
@@ -0,0 +1,231 @@
+      .catch(error => console.log(error));
+    this.props.closeFunction();
+  }
+
+  leaveFocusEmail(): void {
+    this.setState({emailOffFocus: true});
+  }
+
+  enterFocusEmail(): void { this.setState({emailOffFocus: false}); }
+
+  updateContactEmail(evt: React.ChangeEvent<HTMLInputElement>): void {
+    this.setState({contactEmail: evt.target.value});
+  }
+
+  contactEmailInvalidError(): boolean {
+    return !(new RegExp(/^[^\s@]+@[^\s@]+\.[^\s@]{2,}$/).test(this.state.contactEmail));
+  }
+
+  showEmailValidationError(): boolean {
+    if (isBlank(this.state.contactEmail) || !this.state.emailOffFocus) {
+      return false;
+    }
+    return this.contactEmailInvalidError();
+  }
+
+  close(): void {
+    this.props.closeFunction();
+  }
+
+  render() {
+    return <React.Fragment>
+      {this.props.update &&
+      <Modal>
+        <ModalTitle>Change contact email</ModalTitle>
+        <ModalBody>
+          <table style={{width: '100%'}}>
+            <tbody>
+            <tr>
+              <td><label>Contact Email:</label></td>
+              <td style={{width: '70%'}}><FieldInput
+                id='change-contact-email'
+                style={this.showEmailValidationError() ? styles.unsuccessfulInput : {}}
+                onChange={(e) => this.setState({contactEmail: e.target.value})}
+                onBlur={() => this.leaveFocusEmail()}
+                onFocus={() => this.enterFocusEmail()}
+              >
+              </FieldInput>
+              </td>
+            </tr>
+            <tr>
+              <td></td>
+              <td style={{width: '76%'}}>
+                {this.showEmailValidationError() &&
+                <Error>Email is not valid.</Error>
+                }
+              </td>
+            </tr>
+            </tbody>
+          </table>
+        </ModalBody>
+        <ModalFooter>
+          <button type='button' className='btn btn-outline'
+                  onClick={() => this.close()}>Cancel</button>
+          <button id='change_email' type='button'
+                  className='btn btn-primary'
+                  onClick={() => this.updateAndSend()}>Apply</button>
+        </ModalFooter>
+      </Modal>}
+    </React.Fragment>;
+  }
+
+}
+
+@Component({
+  selector: 'app-account-creation-modals',
+  templateUrl: './component.html',
+})
+
+export class AccountCreationModalsComponent implements OnInit, DoCheck {
+  @Input('updateEmail')
+  public updateEmail: Function;
+  @Input('close')
+  public close: Function;
+  @Input('username')
+  public userName: string;
+  @Input('creationNonce')
+  public creationNonce: string;
+  @Input('update')
+  public update: boolean;
+  @Input('resend')
+  public resend: boolean;
+
+  constructor(
+  ) {}
+
+  renderModal(): void {
+    let component;
+    const props = {username: this.userName, creationNonce: this.creationNonce,
+                   closeFunction: this.close};
+    if (this.resend) {
+      component = AccountCreationResendModalReact;
```

Comment:
TOL: Normally we'd have two separate wrapper components here, but that would be overly verbose, so this is fine for now. Once the wrapper is abstracted out we can tighten this up a bit.

Reply:
That's exactly what I was thinking